designdocs/permissions.md

```diff
@@ -0,0 +1,87 @@
+# proposal for permissions request
+
+
+### Philosophy
+The user-agent will always be closer to the user than a spec. The user agent has greater access to current conditions than we (the spec writers) ever will. As a philosophy we should give the user agent *as much freedom as possible to “do the right thing”* under whatever circumstances are at the time a web application is started.
+
+### scope
+
+Currently we are having difficulties finding a solution that all parties can agree to. This is about the permissions for various properties in WebXR.  This proposal addresses the following concerns
+
+* what permissions are available to be asked for
+* how granular are these permissions
+* are the permissions granted at page load / application start
+* or are the permissions granted at the time of use.
+
+### stating concerns
+
+Concerns about granting at page load time are
+* users may be requested to approve multiple permissions one after another, creating click fatigue.
+* if all apps request permissions at start, the user has no information to judge whether or not they should grant this, because they haven’t entered the experience yet.
+
+Concerns about granting requests once already inside of immersive mode are:
```

Comment:
Consider adding an additional concern here that the resulting prompt sequence may be confusing?

For example, if the UA initially creates an AR session with just poses + hit test, it may show a session consent prompt that says "the site does not get access to camera images". If the app then requests camera frames during the session, it would be confusing to pop up a new prompt that contradicts this and asks for site camera access after all.

I like your proposed solution which I think addresses this nicely. If the user agent already knows at session creation time which access may be requested later, it could use this information to avoid confusing prompt sequences.

Reply:
I agree. That's another concern we should add to the doc.